### ISSUE TITLE
Analayzer for misuse of `Arg.Any<T>` in xunits `Asert.Throws<T>`?

### ISSUE THREAD

**Issue by FroggieFrog** [nsubstitute/NSubstitute.Analyzers#209]
Is there already an analyzer that schould trigger when the following incorrect use happens?
This caused some isssues in my test library and I only found out after commenting out about 50 tests. So an analyzer would be great for that kind of scenario.

```csharp
using System;
using NSubstitute;
using Xunit;

namespace Tests;

public interface IMyInterface
{
    int MyMethod(string myParameter);
}

public class TestClass
{
    [Fact]
    public void IStItem_GetNext_ShouldThrow()
    {
        // Arrange
        var substitute = Substitute.For<IMyInterface>();

        // Act/Assert
        Assert.Throws<NotImplementedException>(() => substitute.MyMethod(Arg.Any<string>())); // <-- INcorrect | should trigger
        Assert.Throws<NotImplementedException>(() => substitute.MyMethod(string.Empty)); // <-- correct | should NOT trigger
    }
}
```

**Comment by tpodolak** [comment]
> Is there already an analyzer that schould trigger when the following incorrect use happens?

No, we dont have analyzer for these cases. We actually wanted to have one and attempted to write it but after testing on actual codebases  we ended up with much simplified version which only check usages on non-virtual members. Here are relevant issue and pr with all necessary discussions why we ended up with rather basic analyzer for arg misuse
https://github.com/nsubstitute/NSubstitute.Analyzers/issues/35 
https://github.com/nsubstitute/NSubstitute.Analyzers/pull/120
